server.py

```diff
@@ -29,7 +31,30 @@ def download(number):
     input_file_name = 'inputs'+number+'.txt'
     return static_file(input_file_name, root=dir_path+'/'+'files/questions/')
 
-
-run(app, host = 'localhost', port = 8080)
-
-
+@app.route('/upload', method = 'POST')
+def file_upload():
+    upload = request.files.get('upload')
+    name, ext = os.path.splitext(upload.filename)
+
+    save_path = "./files/user_outputs/"
+    upload.save(save_path)  # appends upload.filename automatically
+    
+    userpath = "./files/user_outputs/" + name + '.' + ext
+
+    with open('./files/expected_outputs/expected_output.txt','r') as f2 , open(userpath,'r') as f1:
+        ans = True
```

Comment:
we can avoid this loop completely.
```suggestion
        ans = f1.read().strip() == f2.read().strip()
```